core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/clazz/list/ClazzListViewModel.kt

```diff
@@ -161,17 +165,23 @@ class ClazzListViewModel(
     }
 
     override fun onClickAdd() {
-        navigateToCreateNew(ClazzEditViewModel.DEST_NAME)
+        navigateToCreateNew(
+            ClazzEditViewModel.DEST_NAME,
+            extraArgs = mapOf(UstadView.ARG_CAN_ADD_COURSE to _uiState.value.canAddNewCourse.toString())
```

Comment:
No! 

ClazzEditViewModel should check permissions to determine what to show - just like every other view model does. It should not be passed from anywhere else. **Revert this change**.

This makes the code less predictable. Adding an extra argument that needs passed to ClassEdit, from every which path someone could get to ClazzEdit. This is against the single point of truth principle.

It's also insecure: someone could simply add the parameter themselves to the URL.